### PR TITLE
Fix `mask` being replaced by the `blind_source_separation` method

### PR DESCRIPTION
**Pull request by francisco-dlp**
### Description of the change

Avoid replacing the original mask by deepcopying it.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: upcoming_changes/3384.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Avoid replacing the original mask in `blind_source_separation`
```
Comment (ericpre) [inline upcoming_changes/3384.bugfix.rst]:
```suggestion
Avoid replacing the original mask in :meth:`~.api.signals.BaseSignal.blind_source_separation`
```

**Review by ericpre — APPROVED** [review]
The decomposition doesn't have this bug but would it be worth updating the corresponding tests (using mask in decomposition) similarly as this is done for bss in this PR?

**Comment by francisco-dlp** [comment]
Thanks for the suggestion @ercipre, done!

**Comment by ericpre** [comment]
pre-commit.ci autofix

**Pull request merged**